tests/functional/ionIonBeam/ion_ion_beam1d.py

```diff
@@ -125,16 +130,19 @@ def growth_b_right_hand(run_path):
         x = xyz_finest[0][:-1]
 
         by = by_interpolator(x)
-        bz = by_interpolator(x)
+        bz = bz_interpolator(x)
 
         # get the mode 1, as it is the most unstable in a box of length 33
         mode1 = np.absolute(np.fft.fft(by-1j*bz)[1])
         first_mode = np.append(first_mode, mode1)
+    
+    imax = find_peaks(first_mode,width = int(d/dt))[0][0]
+    
 
-    popt, pcov = curve_fit(yaebx, times, first_mode, p0=[0.08, 0.09])
+    popt, pcov = curve_fit(yaebx, times[:imax-int(d/dt)], first_mode[:imax-int(d/dt)], p0=[0.08, 0.09])
```

Comment:
add comment
the curve_fit is performed from time index 0 to imax-ioffset as this offset prevent to use the final part of the curve which is no more exponential as this is the end of the linear mode